blog/guide-on-graphiql-2024-07-24.md

```diff
@@ -0,0 +1,176 @@
+
+### Variables Editor
+
+Easily edit variables with autocomplete based on the ones declared in your query—no more tedious copy-pasting of long variable names in complex queries.
+
+![vars screenshot](/images/graphiql/vars.png)
+
+### Response Pane
+
+Check out the server's response here. It's super useful for handling lots of nested data, with built-in folding to keep things tidy and avoid getting lost in a sea of JSON.
+
+![response pane](/images/graphiql/response.png)
+
+## Setting Headers
+
+1. Access the Headers Panel right beside the variables panel at the bottom
+
+![headers screenshot](/images/graphiql/header.png)
+
+2. Add Custom Headers
+
+```json
+{
+  "Authorization": "Token myGhToken"
+}
+```
+
+3. Run Your Query
+
+## Best Practices
+
+### Write Clear and Well-Structured Queries
+
+Keep your queries easy to read and understand. Clean queries make your life (and your future self's) way easier.
+
+**Not Cool**:
+
+```graphql
+{
+  user(id: 1) {
+    id
+    name
+    email
+    friends {
+      id
+      name
+      posts {
+        id
+        title
+        content
+        comments {
+          id
+          text
+        }
+      }
+    }
+  }
+}
+```
+
+**Totally Awesome**:
+
+```graphql
+{
+  user(id: 1) {
+    id
+    name
+    email
+    friends {
+      id
+      name
+      posts {
+        id
+        title
+        content
+        comments {
+          id
+          text
+        }
+      }
+    }
+  }
+}
+```
+
+### Use Modular Queries
+
+Break down those monster queries into smaller, bite-sized pieces. It makes your code look slick and easy to handle.
+
+### Implement Thorough/Consistent Query Testing
+
+Test your queries regularly to catch bugs early. Reliable queries make for a happy developer.
+
+### Add In-Line Comments
+
+Drop some comments in your queries to explain what’s going on, it will make life much easier. Future you (and your teammates) will thank you!
+
+## Alternatives
+
+In case GraphiQL doesn’t quite work out for you and you want to try another IDE, here are some options:
+
+- **[Postman](https://learning.postman.com/docs/sending-requests/graphql/graphql-overview/)**
+- **[Insomnia](https://docs.insomnia.rest/insomnia/graphql-queries)**
+- **[Altair Client](https://altairgraphql.dev/)**
+
+## Conclusion
+
+Wrapping up, we discussed how GraphiQL is an absolute life-saver, especially when you have a server with hundreds of queries returning hundreds of fields of response data. It also makes it super easy to set up a GraphQL playground, like the Tailcall Playground.
+
+It's relieving and exciting to see new open-source projects being launched that make our lives as programmers much easier. Tailcall absolutely boosts server speed and performance, GraphiQL is a total blessing for servers in the development stage, and [Hygraph](https://hygraph.com/) is the go-to for developing CMS quickly.
```

Comment:
Hygraph?

Reply:
Had to make a general statement regarding the topic, so chose to talk about some famous products like hygraph and tailcall.